include/seqan3/core/simd/detail/builtin_simd.hpp

```diff
@@ -0,0 +1,219 @@
+    //!\brief seqan3::simd::simd_traits<builtin_simd_t> should be able to access *scalar_type* and *length*.
+    friend class seqan3::simd::simd_traits<builtin_simd_t>;
+
+    //!\brief The scalar type of builtin_simd_t
+    using scalar_type = std::remove_reference_t<decltype(std::declval<builtin_simd_t>()[0])>;
+    //!\brief The length of builtin_simd_t
+    static constexpr auto length = min_viable_uint_v<sizeof(builtin_simd_t)/sizeof(scalar_type)>;
+
+public:
+    //!\brief Whether builtin_simd_t is a builtin type or not?
+    //!\hideinitializer
+    static constexpr bool value = std::is_same_v<builtin_simd_t, transformation_trait_or_t<builtin_simd<scalar_type, length>, void>>;
+};
+
+/*!\brief seqan3::detail::is_builtin_simd is std::true_type, **iff**
+ * seqan3::detail::builtin_simd<scalar_t, length>::type is a builtin simd type.
+ * \ingroup simd
+ * \tparam scalar_type The underlying type of a simd vector
+ * \tparam length_v The number of packed values in a simd vector
+ *
+ * \include test/snippet/core/simd/detail/is_builtin_simd.cpp
+ * \sa https://gcc.gnu.org/onlinedocs/gcc/Vector-Extensions.html
+ */
+template <typename builtin_simd_t>
+struct is_builtin_simd : std::bool_constant<_is_builtin_simd<builtin_simd_t>::value>
+{};
+
+/*!\brief This function specializes seqan3::detail::default_simd_max_length for
+ * seqan3::detail::builtin_simd.
+ * \ingroup simd
+ *
+ * The redefinition of *default_simd_max_length* influences the default
+ * *length* (i.e., seqan3::detail::default_simd_length) of seqan3::simd::simd_type for
+ * seqan3::detail::builtin_simd types.
+ *
+ * \attention
+ * builtin_simd does not support *avx512* (64 * 8bit = 512bit) and
+ * will fall back to *avx2* (32 * 8bit = 256bit).
+ */
+template <>
+constexpr auto default_simd_max_length<builtin_simd> = []()
+{
+#if defined(__AVX512F__)
+    return min_viable_uint_v<32u>;
+#elif defined(__AVX2__)
+    return min_viable_uint_v<32u>;
+#elif defined(__SSE4_1__) && defined(__SSE4_2__)
+    return min_viable_uint_v<16u>;
+#else
+    // static_assert(false, "Your code uses simd, but you didn't provide flags "
+    // "(like `-march=native`, `-msse4`, `-mavx2`, etc) to your compiler "
+    // "to build special binaries that make use of the special vector/simd "
+    // "registers.");
+    return min_viable_uint_v<0u>;
+#endif
+}();
+
+} // namespace seqan3::detail
+
+namespace seqan3::simd
+{
+
+/*!\brief This class specializes seqan3::simd::simd_traits for seqan3::detail::builtin_simd types
+ * \tparam builtin_simd_t A simd type that satisfies seqan3::detail::is_builtin_simd_v<builtin_simd_t>.
+ * \ingroup simd
+ * \sa seqan3::simd::simd_traits for more information
+ */
+template <typename builtin_simd_t>
+// \cond
+    requires detail::is_builtin_simd<builtin_simd_t>::value
+// \endcond
+struct simd_traits<builtin_simd_t>
+{
+    //!\copydoc seqan3::simd::simd_traits::scalar_type
+    using scalar_type = typename detail::_is_builtin_simd<builtin_simd_t>::scalar_type;
```

Comment:
wait, this is really weird that something that is called `is_...` which implies bool now provides other member types like a type_traits object :thinking: